api/src/index.ts

```diff
@@ -62,6 +63,47 @@ app.post("/registerlead", async (req, res) => {
   }
 });
 
+app.get("/leads/:id", async (req, res) => {
+  try {
+    const connection = getConnection();
+
+    const [details]: any[] = await connection.execute(
+      `SELECT *
+      FROM leads, contactInfo, companyInfo
+      WHERE id = ?`,
+      [req.params.id]
+    );
+
+    const leadDetails: Lead = {
+      id: details[0].id,
```

Comment:
maybe create a new const that will be the first object of the details array and then call it instead of the [0]